Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -631,9 +631,9 @@ int CLuaEngineDefs::EngineFreeModel(lua_State* luaVM)
 
     if (!argStream.HasErrors())
     {
-        CClientModel* pModel = m_pManager->GetModelManager()->FindModelByID(iModelID);
-
-        if (pModel && pModel->Deallocate())
+        auto modelManager = m_pManager->GetModelManager();
+        std::shared_ptr<CClientModel> pModel = modelManager->FindModelByID(iModelID);
+        if (pModel && modelManager->Remove(pModel))
```

Comment:
I'd that adding a function to remove a model by ID would be a better solution.

Reply:
Yes, but `Remove` could do this mess, instead of us.
(Also, its a little slower because of the shared ptr copy, but its negligible)